### ISSUE TITLE
Having alternatives sometimes breaks the parser.

### ISSUE THREAD

**Issue by IanTayler** [IanTayler/MinG#2]
# The issue
This issue is best explained by example (partly because I don't know what's the problem behind the scenes: only that certain grammars I've experimented with don't work). The following grammars don't work for the issues mentioned

```
START=C
(empty) :: =V C
imasentence :: V
comio :: =D =D V
juan :: D
```
This one recognises _juan comio juan_ properly, but it doesn't recognise _imasentence_. The same grammar without the lexical item _comio_ does recognise _imasentence_.
```
START=C
(empty) :: =V C
imasentence :: C # Pay attention to the difference here.
hay :: =D V 
juan :: D
```
This one recognises _hay juan_ but not _imasentence_, even though _imasentence_ is directly specified as a word-sentence (i.e. its feature specification includes only the start category).
```
START=C
(empty) :: =V C
imasentence :: C
hay :: =D V 
juan :: D
comio :: =D =D V
```
In this last one nothing works. Not _imasentence_, not _hay juan_ and not _juan comio juan_. This can be reproduced rather minimally as
```
START=V
comio :: =D =D V
hay :: =D V
juan :: D
```
Which fails to recognise any of the strings in the language.

# Some guesses
It's likely that the issue is closely related (or just essentially the same issue) as #1.

The problem only appears when there's alternatives with different feature-specifications. There is no such problem if in the last example we change "hay :: =D V" for "escupe :: =D =D V". So it's possible something is going wrong with Node.children_with_property() or around that part of the code when it decides whether or not it should scan() (it seems to be getting false negatives there). In some cases, the parser favours the deeper alternative (i.e. not recognising word-sentences while properly recognising intransitive sentences), in some other cases, both options fail.

Let's just hope it isn't an asynchrony problem.

**Comment by IanTayler** [comment]
The issue is much improved now. Most of the problem is gone after I implemented Queue.deep_clone(). As expected there was an important performance drop, which is the price to pay for correct code in this case.

I can't quite close it yet because of the following asymmetry I found in a grammar. Parsing with the following grammar recognises _el muchacho comio la gran mermelada_ but not _el gran muchacho comio la mermelada_ which is rather unexpected behaviour (a bad implementation of SpICm? Why would we have specifier-complement asymmetries otherwise?).
```
START=V
comio       :: =D =D V
el          :: =N D
gran        :: =N N
muchacho    :: N
muchacha    :: N
la          :: =N D
Juan        :: D
mermelada   :: N
```
That looks nothing like issue #1 , so I'm counting it as part of this issue.
